docs/authentication.md

```diff
@@ -0,0 +1,318 @@
+---
+description: Learn about authentication patterns in Next.js apps and explore a few examples.
+---
+
+# Authentication
+
+Authentication verifies who a user is, while authorization controls what a user can access. Next.js supports multiple patterns for authentication, each designed for different use cases. This guide will allow you to choose your adventure based on your constraints.
+
+## Patterns
+
+The first step to identifying which authentication pattern you need is understanding the data-fetching strategy you want. We can then determine which authentication providers support this strategy. There are two main strategies:
+
+- Use static generation to server-render a loading state, followed by fetching user data client-side.
+- Fetch user data server-side to eliminate a flash of unauthenticated content.
```

Comment:
Make sure that every mention to a Next.js feature (data fetching, server rendering, static generation, etc) links to the documentation page that explains it.

Reply:
Added! Let me know if I missed any.